inprocess/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -801,11 +812,6 @@ public void writeMessage(InputStream message) {
           statsTraceCtx.outboundMessageSent(outboundSeqNo, -1, -1);
           serverStream.statsTraceCtx.inboundMessage(outboundSeqNo);
           serverStream.statsTraceCtx.inboundMessageRead(outboundSeqNo, -1, -1);
-          statsTraceCtx.outboundUncompressedSize(messageLength);
```

Comment:
So we're just willy-nilly moving methods out of a synchronized block and after the `closed` check? Since the code is disabled and we want to get this in, I can accept this, but that is a very important detail that should have been called out in the commit/PR description. It is really easy to not notice that when reviewing.

I'm thinking it may actually be safe to do this move, but I'd need to glance at history a bit to see how we ended up with this current construction. Even if it is safe, I don't know whether we want to avoid the lock; we'd probably want to be consistent with the other `statsTraceCtx` (and so change them as well, or not change this).

Reply:
Okay, so I can move it back there inside a synchronized block, but with a check again if (isEnabledSupportTracingMessageSizes()).

Reply:
Your choice.

Reply:
Alright. I added a commit. Please take a final look at the changes before I merge.